src/keri/core/eventing.py

```diff
@@ -4285,6 +4285,7 @@ def processReplyEndRole(self, *, serder, saider, route,
         aid = cid  # authorizing attribution id
         keys = (aid, role, eid)
         osaider = self.db.eans.get(keys=keys)  # get old said if any
+        if osaider is not None and osaider.qb64b == saider.qb64b: osaider = None
```

Comment:
2 things...

1.  please wrap if statements like this otherwise it triggers a PEP warning.
2. We need to discuss this as I believe this breaks BADA RUN logic.

Reply:
1- done
2- Yes, I need your careful review. I think it is not breaking bada run since it's preventing to compare the same said (old = new), so you'll be updating/authorizing the same aid-role-eid value again.